ksqldb-engine/src/main/java/io/confluent/ksql/logicalplanner/nodes/SelectNode.java

```diff
@@ -31,15 +43,41 @@ public SelectNode(
     super(input);
     Objects.requireNonNull(selectClause, "selectClause");
 
-    // expression after check only works with '*''
-    if (selectClause.getSelectItems().size() > 1
-        || !(selectClause.getSelectItems().get(0) instanceof AllColumns)) {
-      throw new UnsupportedOperationException("Only `SELECT *` supported");
-    }
+    final ImmutableList<LogicalColumn> inputSchema = input.getOutputSchema();
+    final Builder<LogicalColumn> outputSchemaBuilder = ImmutableList.builder();
 
-    outputSchema = selectClause.getSelectItems().stream().flatMap(
-        s -> input.getOutputSchema().stream()
-    ).collect(ImmutableList.toImmutableList());
+    selectClause.getSelectItems().forEach(
+        s -> {
+          if (s instanceof AllColumns) {
+            outputSchemaBuilder.addAll(inputSchema);
+          } else {
+            final Expression e = ((SingleColumn) s).getExpression();
+            // to-do: maybe extend using
+            // `UnqualifiedColumnReferenceExp` and `QualifiedColumnReferenceExp`
+            if (e instanceof ColumnReferenceExp) {
+              final ColumnName columnName = ((ColumnReferenceExp) e).getColumnName();
+              final ImmutableList<LogicalColumn> matchingInputColumns = inputSchema.stream()
+                  .filter(logicalColumn -> logicalColumn.name().equals(columnName))
+                  .collect(ImmutableList.toImmutableList());
+              if (matchingInputColumns.isEmpty()) {
+                throw new UnknownColumnException("SELECT", (ColumnReferenceExp) e);
+
+              }
+              if (matchingInputColumns.size() > 1) {
+                throw new KsqlException("Ambiguous column " + columnName + " in SELECT clause");
```

Comment:
It depends. For a "transient query" it does not really matter (but atm the planner is only for persistent queries). For a `CREATE AS` is does matter though because a schema cannot contain overlapping column names (and the `SELECT` defines the schema of the result STREAM/TABLE): it would be required to rename one column using `AS newName` (will as support for `AS` later -- I would assume there is some test already but will double check when working on `AS`).

To be fair, atm ksqlDB has some "auto renaming" feature, but I think it's actually not a good thing to auto-rename and we should rather throw an error and force users to expliclity pick column names IMHO.

Thoughts?